application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationTisStatusUpdateService.java

```diff
@@ -0,0 +1,85 @@
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.service;
+
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.APPROVED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.REJECTED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationStatus.COMPLETED;
+
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.revalidation.dto.RecommendationStatusCheckDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+import uk.nhs.hee.tis.revalidation.repository.RecommendationRepository;
+
+@Slf4j
+@Service
+public class RecommendationTisStatusUpdateService {
+
+  @Autowired
+  private RecommendationRepository recommendationRepository;
+  @Autowired
+  private DoctorsForDBRepository doctorsForDBRepository;
+  @Autowired
+  private RecommendationService recommendationService;
+  @Autowired
+  private SnapshotService snapshotService;
+
+  public void updateRecommendationAndTisStatus(
+      final RecommendationStatusCheckDto recommendationStatusCheckDto) {
+
+    String gmcReferenceNumber = recommendationStatusCheckDto.getGmcReferenceNumber();
+    Optional<DoctorsForDB> optionalDoctorsForDB = doctorsForDBRepository
+        .findById(gmcReferenceNumber);
+    RecommendationGmcOutcome recommendationGmcOutcome = recommendationStatusCheckDto.getOutcome();
+
+    setRecommendationStatusAndUpdateRepositories(gmcReferenceNumber, recommendationGmcOutcome);
+
+    //update the tis status in doctorsfordb to "complete"
+    if (optionalDoctorsForDB.isPresent()) {
+      final var doctorsForDB = optionalDoctorsForDB.get();
+      doctorsForDB.setDoctorStatus(COMPLETED);
+      doctorsForDBRepository.save(doctorsForDB);
+    }
+  }
+
+  //if gmc outcome is approved/rejected, update the relevant recommendation to approved/rejected,
+  //tis status to complete and also update the snapshot repository
+  private void setRecommendationStatusAndUpdateRepositories(final String gmcNumber,
+      final RecommendationGmcOutcome recommendationGmcOutcome) {
+
+    final var recommendations = recommendationRepository.findByGmcNumber(gmcNumber);
+    recommendations.stream().forEach(recommendation -> {
```

Comment:
We just want to update the returned recommendation, shouldn't update all recommendations of that trainee.